### PR TITLE
Managed Identity Azure VM Sample

### PR DESCRIPTION
**Pull request by Robbie-Microsoft**
Successfully ran sample on an Azure VM

### CODE REVIEW

File: lib/msal-node/src/index.ts [hunk 1]
```diff
@@ -36,6 +36,8 @@ export { UsernamePasswordClient } from "./client/UsernamePasswordClient.js";
 
 export {
     Configuration,
+    ManagedIdentityConfiguration,
+    ManagedIdentityIdParams,
```
Comment (Robbie-Microsoft) [inline lib/msal-node/src/index.ts:40]: This is a bugfix needed for the sample to run. These should have already been exported here.

Comment (bgavrilMS) [reply lib/msal-node/src/index.ts:40]: Does this mean we need to release another MSAL preview @Robbie-Microsoft ?

File: samples/msal-node-samples/managed-identity/README.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Managed Identity for Azure VM Sample
```
Comment (bgavrilMS) [inline samples/msal-node-samples/managed-identity/README.md:1]: Let's move away from samples in the MSAL repo. Can we please have a sample here: https://github.com/Azure-Samples/msal-managed-identity/tree/main/src

You can keep this sample here as a "dev app"

File: samples/msal-node-samples/managed-identity/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig to read more about this file */
+
+    /* Projects */
+    // "incremental": true,                              /* Save .tsbuildinfo files to allow for incremental compilation of projects. */
```
Comment (bgavrilMS) [inline samples/msal-node-samples/managed-identity/tsconfig.json]: That's a lot of dead code. Is it needed?

File: samples/msal-node-samples/managed-identity/README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+This sample demonstrates how to use [managed identity via the msal-node library](/lib/msal-node/docs/managed-identity.md) to retrieve tokens for a managed identity application running on an Azure VM.
+
+## Note
+
+This sample is written in TypeScript and was developed with Node version 18.17.0.
```
Comment (bgavrilMS) [inline samples/msal-node-samples/managed-identity/README.md]: Pls add a note that the functionality is in preview (alpha package)

**Review by bgavrilMS — APPROVED** [review]

File: lib/msal-node/src/index.ts (hunk 1 above)
Comment (Robbie-Microsoft) [reply lib/msal-node/src/index.ts:40]: No. Resolving per our offline discussion.

File: samples/msal-node-samples/managed-identity/tsconfig.json (hunk 3 above)
Comment (Robbie-Microsoft) [reply samples/msal-node-samples/managed-identity/tsconfig.json]: Nope, it was auto-generated. I can delete it.

File: samples/msal-node-samples/managed-identity/README.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# Managed Identity for Azure VM Sample
+
+This sample demonstrates how to use [managed identity via the msal-node library](/lib/msal-node/docs/managed-identity.md) to retrieve tokens for a managed identity application running on an Azure VM.
+
+## Note
+
+-   The functionality for this sample is in preview (alpha)
+-   This sample is written in TypeScript and was developed with Node version 18.17.0.
+
+## Setup
```
Comment (pmaytak) [inline samples/msal-node-samples/managed-identity/README.md:10]:
```suggestion
## Project setup
```

File: samples/msal-node-samples/managed-identity/README.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+## Local Changes
+
+Before running the sample on the Azure VM, the userAssignedClientId value in the managedIdentityIdParams object in index.ts needs to be replaced by the client id of the user assigned managed identity that was created in the previous step:
+
+```typescript
+const managedIdentityIdParams: ManagedIdentityIdParams = {
+    userAssignedClientId: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
+};
+```
```
Comment (pmaytak) [inline samples/msal-node-samples/managed-identity/README.md:30]:
```suggestion
## Virtual machine setup

Follow [this guide](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/qs-configure-portal-windows-vm) to setup an Azure VM, as well as add a system assigned and user assigned managed identity to the Azure VM.

## Project setup

In a terminal, navigate to the directory where `package.json` resides. Then type:

```console
    npm install
```

Before running the sample on the Azure VM, the `userAssignedClientId` in the `managedIdentityIdParams` object in `index.ts` needs to be replaced by the client ID of the user assigned managed identity that was created in the previous section:

```typescript
const managedIdentityIdParams: ManagedIdentityIdParams = {
    userAssignedClientId: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
};
```
```

File: samples/msal-node-samples/managed-identity/README.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    npm install
+```
+
+## Register
+
+Follow [this guide](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/qs-configure-portal-windows-vm) to setup an Azure VM, as well as add a system assigned and user assigned managed identity to the Azure VM.
+
+## Local Changes
+
+Before running the sample on the Azure VM, the userAssignedClientId value in the managedIdentityIdParams object in index.ts needs to be replaced by the client id of the user assigned managed identity that was created in the previous step:
```
Comment (pmaytak) [inline samples/msal-node-samples/managed-identity/README.md]: The GitHub suggestion feature doesn't work well... My suggestion is to combine Setup, Register, and Local Changes sections into this:



## Virtual machine setup

Follow [this guide](https://learn.microsoft.com/en-us/entra/identity/managed-identities-azure-resources/qs-configure-portal-windows-vm) to setup an Azure VM, as well as add a system assigned and user assigned managed identity to the Azure VM.

## Project setup

In a terminal, navigate to the directory where `package.json` resides. Then type:

```console
    npm install
```

Before running the sample on the Azure VM, the `userAssignedClientId` in the `managedIdentityIdParams` object in `index.ts` needs to be replaced by the client ID of the user assigned managed identity that was created in the previous section:
```typescript
const managedIdentityIdParams: ManagedIdentityIdParams = {
    userAssignedClientId: "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
};
```

**Review by pmaytak — APPROVED** [review]

**Pull request merged**